### PR TITLE
BSTOR-8 click on a brand -- Part 2

### PR DESCRIPTION
**Pull request by Pacane**
Changed the text of the products based on the selected brand
Using eventBus to change the theme now.

### CODE REVIEW

File: client/src/main/java/com/arcbees/beeshop/client/events/BrandChangedEvent.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package com.arcbees.beeshop.client.events;
```
Comment (christiangoudreau) [inline client/src/main/java/com/arcbees/beeshop/client/events/BrandChangedEvent.java:1]: Don't forget oss header

Comment (Pacane) [inline client/src/main/java/com/arcbees/beeshop/client/events/BrandChangedEvent.java:1]: Ok, I'll create a task to add it to every file that needs it. Will do the actual work in another PR.

File: client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.java [hunk 2]
```diff
@@ -56,50 +62,70 @@
...
+            AppMessages appMessages) {
         this.placeManager = placeManager;
         this.tokenFormatter = tokenFormatter;
+        this.appMessages = appMessages;
+
+        products = new ArrayList<>();
 
         ProductDto shirt = new ProductDto();
-        shirt.setBrandName("Bee");
+        shirt.setBrandName(appMessages.brandName(Brand.ARCBEES));
```
Comment (jclariviere) [inline client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.java]: Brand.default()

File: client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.java [hunk 3]
```diff
@@ -56,50 +62,70 @@
...
         shirt.setPrice(55);
         shirt.setProductName("Shirt");
 
         ProductDto bag = new ProductDto();
-        bag.setBrandName("Bee");
+        bag.setBrandName(appMessages.brandName(Brand.ARCBEES));
```
Comment (jclariviere) [inline client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.java]: Brand.default()

File: client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.java [hunk 4]
```diff
@@ -56,50 +62,70 @@
...
     }
+
+    @Override
+    public void changeBrand(Brand brand) {
+        for (PricePresenter product : products) {
+            ProductDto value = product.getValue();
+
+            value.setBrandName(appMessages.brandName(brand));
+
+            product.setValue(value);
```
Comment (jclariviere) [inline client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.java:102]: @Chris-V @meriouma We're open for ideas on how to handle this.

File: client/src/main/resources/com/arcbees/beeshop/BeeShopCore.gwt.xml [hunk 5]
```diff
@@ -11,8 +11,14 @@
 
     <extend-configuration-property name="gin.ginjector.modules" value="com.arcbees.beeshop.client.ClientModule"/>
 
+
```
Comment (jclariviere) [inline client/src/main/resources/com/arcbees/beeshop/BeeShopCore.gwt.xml]: rm

File: client/src/test/java/com/arcbees/beeshop/client/application/CurrentBrandImplTest.java [hunk 6]
```diff
@@ -31,25 +41,24 @@ public void getCurrentBrand_returnsCurrentBrand() {
...
-    public void updateBrand_changesThemeWhenBrandChanges() {
+    public void updateBrand_doesNotFireEventWhenBrandIsSame() {
         Brand brand = Brand.CHOSEN;
+        currentBrand.update(brand);
+        Mockito.reset(eventBus);
 
         currentBrand.update(brand);
 
-        verify(themeChanger).changeBrand(brand);
+        verify(eventBus, never()).fireEventFromSource(isA(BrandChangedEvent.class), same(currentBrand));
```
Comment (jclariviere) [inline client/src/test/java/com/arcbees/beeshop/client/application/CurrentBrandImplTest.java]: `verifyZeroInteractions()` would be better here.

File: common/src/main/java/com/arcbees/beeshop/common/dto/Brand.java [hunk 7]
```diff
@@ -22,7 +22,7 @@ public static Brand createFromValue(String value) {
             }
         }
 
-        return ARCBEES;
+        return null;
```
Comment (jclariviere) [inline common/src/main/java/com/arcbees/beeshop/common/dto/Brand.java:25]: Why? `return default()` wouldn't be better?

**Comment by jclariviere** [comment]
LGTM when comments are fixed

File: client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.java (hunk 4 above)
Comment (meriouma) [inline client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.java:102]: Use an event. You shouldn't have presenter references in the view I believe

Comment (meriouma) [inline client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.java:102]: It's already an event, have your PricePresenter listen for the BrandChanged event, and I believe you should use slots for the products

Comment (Pacane) [inline client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.java:102]: Alright, let me try to refactor this in another PR. I'll close these right now.

**Pull request merged**